gemfire-core/src/main/java/com/gemstone/gemfire/internal/LinuxSystemStats.java

```diff
@@ -263,10 +257,25 @@ private static void checkOffset(String name, int offset) {
                             f.createLongCounter("diskTime",
                                                 "The total number of milliseconds that measures both completed disk operations and any accumulating backlog of in progress ops.",
                                                 "milliseconds"),
-                            f.createLongGauge("threadsMax",
-                                "The OS limit of maximum number of threads in current session",
+                            f.createLongGauge("totalFileDescriptors",
+                                "The total number of open file handles in this computer " +
+                                    "at the time of data collection. Notice that this is an " +
+                                    "instantaneous count, not an average over the time interval.",
+                                "fds"),
+                            f.createLongGauge("fileDescriptorsSystemMax",
+                                "The OS limit of maximum number of open file handles in this computer.",
+                                "fds"),
+                            f.createLongGauge("totalThreads",
+                                "The total number of threads in the computer at the time of " +
+                                    "data collection. Notice that this is an instantaneous " +
+                                    "count, not an average over the time interval.",
+                                "threads"),
+                            f.createLongGauge("threadsSessionMax",
+                                "The OS limit of maximum number of threads+processes in current session.",
+                                "threads"),
+                            f.createLongGauge("threadsSystemMax",
+                                "The OS limit of maximum number of threads+processes in this computer.",
```

Comment:
The threads and file handles of this JVM are already captured in VMStats using JDK management API (+ reflection for unix specific ones). See VMStats50.threadsId, threadStartsId, unix_fdLimitId, unix_fdsOpenId